astropy/wcs/src/astropy_wcs.c

```diff
@@ -750,47 +751,27 @@ static PyMethodDef module_methods[] = {
   {NULL}  /* Sentinel */
 };
 
-static PyTypeObject WcsType = {
-  PyVarObject_HEAD_INIT(NULL, 0)
-  "astropy.wcs.WCSBase",                 /*tp_name*/
-  sizeof(Wcs),                /*tp_basicsize*/
-  0,                            /*tp_itemsize*/
-  (destructor)Wcs_dealloc,    /*tp_dealloc*/
-  0,                            /*tp_print*/
-  0,                            /*tp_getattr*/
-  0,                            /*tp_setattr*/
-  0,                            /*tp_compare*/
-  0,                            /*tp_repr*/
-  0,                            /*tp_as_number*/
-  0,                            /*tp_as_sequence*/
-  0,                            /*tp_as_mapping*/
-  0,                            /*tp_hash */
-  0,                            /*tp_call*/
-  0,                            /*tp_str*/
-  0,                            /*tp_getattro*/
-  0,                            /*tp_setattro*/
-  0,                            /*tp_as_buffer*/
+static PyType_Slot WCSBaseClass_slots[] = {
+  {Py_tp_dealloc, (destructor)Wcs_dealloc},
+  {Py_tp_doc, doc_Wcs},
+  {Py_tp_traverse, (traverseproc)Wcs_traverse},
+  {Py_tp_clear, (inquiry)Wcs_clear},
+  {Py_tp_methods, Wcs_methods},
+  {Py_tp_getset, Wcs_getset},
+  {Py_tp_init, (initproc)Wcs_init},
+  {Py_tp_new, Wcs_new},
+  {0, NULL},
+};
+
+static PyType_Spec WCSBaseClass_spec = {
+  "astropy.wcs.WCSBase",    /*tp_name*/
+  sizeof(Wcs),              /*tp_basicsize*/
+  0,                        /*tp_itemsize*/
   Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE | Py_TPFLAGS_HAVE_GC, /*tp_flags*/
```

Comment:
I'd also probably add `Py_TPFLAGS_IMMUTABLETYPE` to this, just because static types are immutable and heap types aren't by default. Which may be fine, but I assume you're trying to change as little as possible right now.

Reply:
reading this again, I'm seeing another possible interpretation. Did you mean to suggest `Py_TPFLAGS_IMMUTABLETYPE` should be added *in order* to change as little as possible ?

Reply:
Yes I am.

Reply:
thanks ! I totally read that wrong